src/physicsml/models/ani/modules/aev.py

```diff
@@ -0,0 +1,709 @@
+
+    Output: indices for all central atoms and it pairs of neighbors. For
+    example, if input has pair (0, 1), (0, 2), (0, 3), (0, 4), (1, 2),
+    (1, 3), (1, 4), (2, 3), (2, 4), (3, 4), then the output would have
+    central atom 0, 1, 2, 3, 4 and for cental atom 0, its pairs of neighbors
+    are (1, 2), (1, 3), (1, 4), (2, 3), (2, 4), (3, 4)
+    """
+    # convert representation from pair to central-others
+    ai1 = atom_index12.view(-1)
+    sorted_ai1, rev_indices = ai1.sort()
+
+    # sort and compute unique key
+    uniqued_central_atom_index, counts = torch.unique_consecutive(
+        sorted_ai1,
+        return_inverse=False,
+        return_counts=True,
+    )
+
+    # compute central_atom_index
+    pair_sizes = torch.div(counts * (counts - 1), 2, rounding_mode="trunc")
+    pair_indices = torch.repeat_interleave(pair_sizes)
+    central_atom_index = uniqued_central_atom_index.index_select(0, pair_indices)
+
+    # do local combinations within unique key, assuming sorted
+    m = counts.max().item() if counts.numel() > 0 else 0
+    n = pair_sizes.shape[0]
+    intra_pair_indices = (
+        torch.tril_indices(m, m, -1, device=ai1.device).unsqueeze(1).expand(-1, n, -1)
+    )
+    mask = (
+        torch.arange(intra_pair_indices.shape[2], device=ai1.device)
+        < pair_sizes.unsqueeze(1)
+    ).flatten()
+    sorted_local_index12 = intra_pair_indices.flatten(1, 2)[:, mask]
+    sorted_local_index12 += cumsum_from_zero(counts).index_select(0, pair_indices)
+
+    # unsort result from last part
+    local_index12 = rev_indices[sorted_local_index12]
+
+    # compute mapping between representation of central-other to pair
+    n = atom_index12.shape[1]
+    sign12 = ((local_index12 < n).to(torch.int8) * 2) - 1
+    return central_atom_index, local_index12 % n, sign12
+
+
+def compute_aev(
+    species: Tensor,
+    coordinates: Tensor,
+    triu_index: Tensor,
+    constants: Tuple[float, Tensor, Tensor, float, Tensor, Tensor, Tensor, Tensor],
+    sizes: Tuple[int, int, int, int, int],
+    cell_shifts: Optional[Tuple[Tensor, Tensor]],
+) -> Tensor:
+    Rcr, EtaR, ShfR, Rca, ShfZ, EtaA, Zeta, ShfA = constants
+    (
+        num_species,
+        radial_sublength,
+        radial_length,
+        angular_sublength,
+        angular_length,
+    ) = sizes
+    num_molecules = species.shape[0]
+    num_atoms = species.shape[1]
+    num_species_pairs = angular_length // angular_sublength
+    coordinates_ = coordinates
+    coordinates = coordinates_.flatten(0, 1)
+
+    # PBC calculation is bypassed if there are no shifts
+    if cell_shifts is None:
+        atom_index12 = neighbor_pairs_nopbc(species == -1, coordinates_, Rcr)
+        selected_coordinates = coordinates.index_select(0, atom_index12.view(-1)).view(
+            2,
+            -1,
+            3,
+        )
+        vec = selected_coordinates[0] - selected_coordinates[1]
+    else:
+        cell, shifts = cell_shifts
+
+        # solve pos = coefs^T @ cell
```

Comment:
@wiederm here is the part that changed

Reply:
Oh, I see!

So, just to make sure I understand this overall: you have changed the coordinate wrapping the the `neighbourhood_list_torch.py`, which is the neighbourlist implementation you are using for the NNPs with the exception of ANI?

Reply:
I think this is a possible position within the code to do the wrapping.

But, since this forced you to add the `aev.py` module I think I would have preferred to do the wrapping before passing it to the `compute_aev` function, depending on the cell_shifts variable. But I understand that this might introduce complexity elsewhere.

Reply:
Yeah exactly. 

And for ANI, it has to be inside the `compute_aev` because the cell shifts are computed there, so we have to compute the modifications to those as well.